### PR TITLE
Fix Flashlight IL Error && Fix infinite connect && Ammo drop in ChangingRole && EndingRound NWPluginAPI Fix

### PR DESCRIPTION
**Pull request by louis1706**

### CODE REVIEW

File: Exiled.Events/EventArgs/Player/TogglingFlashlightEventArgs.cs [hunk 1]
```diff
@@ -31,33 +33,32 @@ public class TogglingFlashlightEventArgs : IPlayerEvent, IDeniableEvent
...
 
         /// <summary>
         ///     Gets the <see cref="API.Features.Items.Flashlight" /> being toggled.
         /// </summary>
         public Flashlight Flashlight { get; }
 
-#pragma warning disable SA1623 // Property summary documentation should match accessors
         /// <summary>
-        ///     Gets or sets the new flashlight state.
+        ///     Gets or sets a value indicating whether is will be the NewState of the Light.
```
Comment (Misaka-ZeroTwo) [inline Exiled.Events/EventArgs/Player/TogglingFlashlightEventArgs.cs:50]:
```suggestion
        ///     Gets or sets the new flashlight state.
```

Comment (Misaka-ZeroTwo) [reply Exiled.Events/EventArgs/Player/TogglingFlashlightEventArgs.cs:50]: I don't see a need to change that, plus the change itself is grammatically wrong.

Comment (Misaka-ZeroTwo) [reply Exiled.Events/EventArgs/Player/TogglingFlashlightEventArgs.cs]: 1

File: Exiled.Events/EventArgs/Player/TogglingFlashlightEventArgs.cs [hunk 2]
```diff
@@ -31,33 +33,32 @@ public class TogglingFlashlightEventArgs : IPlayerEvent, IDeniableEvent
...
         }
 
         /// <summary>
         ///     Gets the <see cref="API.Features.Items.Flashlight" /> being toggled.
         /// </summary>
         public Flashlight Flashlight { get; }
 
-#pragma warning disable SA1623 // Property summary documentation should match accessors
         /// <summary>
         ///     Gets or sets the new flashlight state.
```
Comment (Misaka-ZeroTwo) [inline Exiled.Events/EventArgs/Player/TogglingFlashlightEventArgs.cs:50]:
```suggestion
        ///     Gets or sets a value indicating whether or not the flashlight should be on.
```

Comment (Misaka-ZeroTwo) [reply Exiled.Events/EventArgs/Player/TogglingFlashlightEventArgs.cs:50]: Better? @Thundermaker300 insight?

**Review by Misaka-ZeroTwo — COMMENTED** [review]
I guess if you want to change it, here's what I would do.

File: Exiled.Events/Patches/Events/Player/ChangingRoleAndSpawned.cs [hunk 3]
```diff
@@ -246,11 +250,14 @@ private static void ChangeInventory(ChangingRoleEventArgs ev)
...
 
+                foreach (ItemType item in ev.Items)
+                    inventory.ServerAddItem(item);
+
                 foreach (KeyValuePair<ItemType, ushort> keyValuePair in ev.Ammo)
                     inventory.ServerAddAmmo(keyValuePair.Key, keyValuePair.Value);
 
-                foreach (ItemType item in ev.Items)
-                    InventoryItemProvider.OnItemProvided?.Invoke(ev.Player.ReferenceHub, inventory.ServerAddItem(item));
+                foreach (var item in inventory.UserInventory.Items)
```
Comment (iopietro) [inline Exiled.Events/Patches/Events/Player/ChangingRoleAndSpawned.cs:259]:
```suggestion
                foreach (var item in inventory.UserInventory.Items)
```
Don't use `var`.

**Review by iopietro — APPROVED** [review]

File: Exiled.Events/Patches/Events/Player/ChangingRoleAndSpawned.cs (hunk 3 above)
Comment (louis1706) [reply Exiled.Events/Patches/Events/Player/ChangingRoleAndSpawned.cs:259]: Oups

**Comment by louis1706** [comment]
my last commit should be change at IL code later

**Review by Thundermaker300 — APPROVED** [review]

**Pull request merged**